core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -691,6 +697,37 @@ wasm_interp_get_frame_ref(WASMInterpFrame *frame)
 #define read_leb_mem_offset(p, p_end, res) read_leb_uint32(p, p_end, res)
 #endif
 
+#if WASM_ENABLE_MULTI_MEMORY != 0
+/* First read the alignment, then if it has flag indicating following memidx,
+ * read it and set the memory instance to memories[memidx] for following memory
+ * access and boundary check, if it doesn't have flag reset the memory instance
+ * to the default memories[0] */
```

Comment:
it's a bit surprising api to me.
isn't it simpler to introduce macros like read_memarg/skip_memarg?

Reply:
`read_memarg` does sound better than `read_leb_align` , but still need `read_leb_memidx` for some bulk memory opcodes